src/components/TimeseriesChart/TimeseriesChart.test.tsx

```diff
@@ -104,3 +105,37 @@ describe('TimeseriesChart', () => {
     expect(wrapper.find('.context-container').exists()).toBeTruthy();
   });
 });
+
+it('cursor overview renders with ruler specified', async () => {
+  const props = {
+    timeseriesIds: [timeseriesListV2[0].id],
+    ruler: {
+      visible: true,
+      yLabel: (point: ChartRulerPoint) => `${Number(point.value).toFixed(3)}`,
+      timeLabel: (point: ChartRulerPoint) =>
+        new Date(point.timestamp).toISOString(),
+    },
+  };
+  const wrapper = mount(
+    <ClientSDKProvider client={sdk}>
+      <TimeseriesChart {...props} />
+    </ClientSDKProvider>
+  );
+  await sleep(300);
+  wrapper.update();
+  expect(wrapper.find(CursorOverview).exists()).toBeTruthy();
+});
+
+it('cursor overview does not render when ruler is not specified', async () => {
```

Comment:
this test is redundant for my mind

Reply:
This negative test is intended to cover the code that removes the `CursorOverview` when `ruler` prop is absent.

Reply:
I replaced it with one row in init test